src/main/java/com/amazon/ion/impl/bin/IonRawBinaryWriter_1_1.kt

```diff
@@ -299,23 +301,20 @@ class IonRawBinaryWriter_1_1 internal constructor(
     }
 
     /**
-     * Helper function for finishing scalar values. Similar concerns for containers are handled in [stepOut].
+     * Helper function for writing scalar values that builds on [openValue] and also includes updating
+     * the length of the current container.
+     *
+     * @param valueWriterExpression should be a function that writes the scalar value to the buffer, and
+     *                              returns the number of bytes that were written.
      */
-    private inline fun closeScalar(valueWriterExpression: () -> Int) {
+    private inline fun writeScalar(valueWriterExpression: () -> Int) = openValue {
         val numBytesWritten = valueWriterExpression()
         currentContainer.length += numBytesWritten
     }
 
-    /**
-     * Helper function for writing scalar values that composes both [openValue] and [closeScalar].
-     */
-    private inline fun writeScalar(valueWriterExpression: () -> Int) {
-        openValue { closeScalar(valueWriterExpression) }
-    }
-
```

Comment:
🗺️ There really was no benefit in having this be two separate functions. No functional changes here.